tensorflow_probability/python/experimental/mcmc/streaming_reductions_test.py

```diff
@@ -0,0 +1,167 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Tests for drivers in a Streaming Reductions Framework."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+from absl.testing import parameterized
+import collections
+import warnings
+
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+from tensorflow_probability.python.internal import samplers
+from tensorflow_probability.python.internal import test_util
+from tensorflow_probability.python.experimental.mcmc.streaming_reductions import step_kernel
+
+
+TestTransitionKernelResults = collections.namedtuple(
+    'TestTransitionKernelResults', 'counter_1, counter_2')
+
+
+class TestTransitionKernel(tfp.mcmc.TransitionKernel):
+  """Fake deterministic `TransitionKernel` for testing purposes"""
+
+  def __init__(self, is_calibrated=True, accepts_seed=True):
+    self._is_calibrated = is_calibrated
+    self._accepts_seed = accepts_seed
+
+  def one_step(self, current_state, previous_kernel_results, seed=None):
+    if seed is not None and not self._accepts_seed:
+      raise TypeError('seed arg not accepted')
+    return current_state + 1, TestTransitionKernelResults(
+        counter_1=previous_kernel_results.counter_1 + 1,
+        counter_2=previous_kernel_results.counter_2 + 2)
+
+  def bootstrap_results(self, current_state):
+    return TestTransitionKernelResults(counter_1=0, counter_2=0)
+
+  @property
+  def is_calibrated(self):
+    return self._is_calibrated
+
+
+class RandomTransitionKernel(tfp.mcmc.TransitionKernel):
+  """Fake `TransitionKernel` that randomly assigns the next state.
+
+  Regardless of the current state, the `one_step` method will always
+  randomly sample from a Reyleigh Distribution.
+  """
+
+  def __init__(self, is_calibrated=True, accepts_seed=True):
+    self._is_calibrated = is_calibrated
+    self._accepts_seed = accepts_seed
+
+  def one_step(self, current_state, previous_kernel_results, seed=None):
+    if seed is not None and not self._accepts_seed:
+      raise TypeError('seed arg not accepted')
+    random_next_state = tfp.random.rayleigh((1,), seed=seed)
+    return random_next_state, previous_kernel_results
+
+  @property
+  def is_calibrated(self):
+    return self._is_calibrated
+
+
+@test_util.test_all_tf_execution_regimes
+class StepKernelTest(test_util.TestCase):
+
+  def test_simple_operation(self):
+    fake_kernel = TestTransitionKernel()
+    final_state, kernel_results = step_kernel(
+        num_steps=2,
+        current_state=0,
+        kernel=fake_kernel,
+        return_final_kernel_results=True
+    )
+    final_state = self.evaluate(final_state)
+    self.assertEqual(final_state, 2)
```

Comment:
This test doesn't actually pass, at least internally.  The error is
```
Traceback (most recent call last):
  File "/build/work/3186db25de8c50f31eeb96306c7279a3f85c/google3/runfiles/google3/third_party/py/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/build/work/3186db25de8c50f31eeb96306c7279a3f85c/google3/runfiles/google3/third_party/py/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/build/work/3186db25de8c50f31eeb96306c7279a3f85c/google3/runfiles/google3/third_party/py/absl/testing/parameterized.py", line 282, in bound_param_test
    return test_method(self, **testcase_params)
  File "/build/work/3186db25de8c50f31eeb96306c7279a3f85c/google3/runfiles/google3/third_party/py/tensorflow_probability/python/internal/test_combinations.py", line 307, in decorated
    execute_test_method()
  File "/build/work/3186db25de8c50f31eeb96306c7279a3f85c/google3/runfiles/google3/third_party/py/tensorflow_probability/python/internal/test_combinations.py", line 295, in execute_test_method
    test_method(**kwargs_to_pass)
  File "/build/work/3186db25de8c50f31eeb96306c7279a3f85c/google3/runfiles/google3/third_party/py/tensorflow_probability/python/experimental/mcmc/streaming_reductions_test.py", line 88, in test_simple_operation
    self.assertEqual(kernel_results.counter_1, 2)
  File "<embedded stdlib>/unittest/case.py", line 513, in assertEqual
    assertion_func(first, second, msg=msg)
  File "<embedded stdlib>/unittest/case.py", line 506, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: <tf.Tensor: shape=(), dtype=int32, numpy=2> != 2
```

Presumably because a Tensor is not actually equal to 2.  Do you also see that on your end?  Should be possible to fix it by using assertAllEqual and/or tf.constant(2, dtype=tf.int32).

Reply:
Talked to Alexey about this offline. There's some discrepancy between BUILDs that makes me unable to reproduce these errors. We concluded that it wouldn't be worthwhile for me to try to get around this (i.e. by building TensorFlow from source). Alexey will handle these errors in the pull in (thanks Alexey!)